crates/oxc_linter/src/external_plugin_store.rs

```diff
@@ -79,8 +79,11 @@ impl ExternalPluginStore {
         })
     }
 
-    pub fn resolve_plugin_rule_names(&self, external_rule_id: u32) -> Option<(&str, &str)> {
-        let external_rule = self.rules.get(ExternalRuleId::from_raw(external_rule_id))?;
+    pub fn resolve_plugin_rule_names(
+        &self,
+        external_rule_id: ExternalRuleId,
+    ) -> Option<(&str, &str)> {
+        let external_rule = self.rules.get(external_rule_id)?;
```

Comment:
if the argument here is `ExternalRuleId`, I don't think we need to use `.get` since we're indexing into the index vec with the correct key

Reply:
the reason this returned an option originally is because you could pass in an arbitary `u32` value which might not be in the correct range.

(i guess you could pass in an arbitary value for `ExternalRuleId`, but it's a lot less likley

Reply:
Good point. While there's no real difference between `u32` and `ExternalRuleId`, the ID now comes from within Rust code, rather than being passed from JS, so we can have more confidence that's it's valid.

I've pushed a commit to make `resolve_plugin_rule_names` infallible.